src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -212,14 +245,34 @@ const UsersFragment = ({
       ? withoutRecent
         ? initialUsers.length !== 0
           ? initialUsers
-          : [...initFollowingUsers, {separator: true}, ...initUnfollowingUsers]
-        : [...initFollowingUsers, {separator: true}, ...initUnfollowingUsers]
+          : [{separator: true}, ...initUnfollowingUsers]
+        : [{separator: true}, ...initUnfollowingUsers]
+      : unfollowedUsers.length !== 0
+      ? [{separator: true}, ...unfollowedUsers]
+      : [];
+
+    const firstData = isFirstTimeOpen
+      ? withoutRecent
+        ? initialUsers.length !== 0
+          ? initialUsers
+          : [...initFollowingUsers]
+        : [...initFollowingUsers]
       : unfollowedUsers.length !== 0
-      ? [...followedUsers, {separator: true}, ...unfollowedUsers]
+      ? [...followedUsers]
       : followedUsers;
 
     return (
       <FlatList
+        ListHeaderComponent={() => (
+          <>
+            <RecentSearch
+              shown={showRecentSearch || isFirstTimeOpen}
+              setSearchText={setSearchText}
+              setIsFirstTimeOpen={setIsFirstTimeOpen}
+            />
+            <AccordionView data={firstData} />
+          </>
+        )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There's a lot of ternary operators here which makes the code hard to read and understand. Consider refactoring this to use if-else statements or a function that returns the desired value based on conditions. This will improve readability and maintainability of your code.

```diff
-    const firstData = isFirstTimeOpen
-      ? withoutRecent
-        ? initialUsers.length !== 0
-          ? initialUsers
-          : [...initFollowingUsers]
-        : [...initFollowingUsers]
-       : unfollowedUsers.length !== 0
-      ? [...followedUsers]
-       : followedUsers;
+    let firstData = followedUsers;
+    if (isFirstTimeOpen) {
+      firstData = withoutRecent && initialUsers.length !== 0 ? initialUsers : [...initFollowingUsers];
+    } else if (unfollowedUsers.length !== 0) {
+      firstData = [...followedUsers];
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->